protos/sigstore_verification.proto

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package dev.sigstore.verification.v1;
+
+import "sigstore_trustroot.proto";
+import "sigstore_bundle.proto";
+
+option go_package = "github.com/sigstore/proto/verification/v1";
+option java_package = "dev.sigstore.proto.verification.v1";
+option java_multiple_files = true;
+option java_outer_classname = "VerificationProto";
+
+enum SubjectAlternativeNameType {
+        DNS = 0;
+        Email = 1;
+        URI = 2;
+}
+
+message SubjectAlternativeName {
+        SubjectAlternativeNameType type = 1;
+        // A regular expression describing the expected value for the SAN.
+        string value = 2;
+}
+
+// The identity of a signer.
+message Identity {
+        // The X.509v3 issuer extension (OID 1.3.6.1.4.1.57264.1.1)
+        string issuer = 1;
+        oneof identity {
+                SubjectAlternativeName san = 2;
+        }
+}
+
+// A light-weight set of options/policies for identifying trusted signers,
+// used during verification of a single artifact.
+message ArtifactVerificationOptions {
+        message TlogOptions {
+                // Number of transparency logs the entry must appear on.
+                int32 threshold = 1;
+                // Perform an online inclusion proof.
+                // TODO: @haydentherapper: wanna punt this for now?
+                bool perform_online_verification = 2;
```

Comment:
@asraa Do you think we're ready for this? I think yes, given our conversation about two flags, one for online, one for offline.

Reply:
I think we can include this. We know it's going to be there, and it's safe for clients to ignore.

Reply:
Seems to be a consencus, resolving.